pkg/cmd/server/options.go

```diff
@@ -14,51 +14,75 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// Package server provides configuration options for the metrics API server.
 package server
 
 import (
 	"fmt"
 	"net"
 
+	"github.com/spf13/pflag"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	genericapiserver "k8s.io/apiserver/pkg/server"
 	genericoptions "k8s.io/apiserver/pkg/server/options"
 	openapicommon "k8s.io/kube-openapi/pkg/common"
 
 	"sigs.k8s.io/custom-metrics-apiserver/pkg/apiserver"
 )
 
+// CustomMetricsAdapterServerOptions contains the of options used to configure
+// the metrics API server.
+//
+// It is based on a subset of [genericoptions.RecommendedOptions].
 type CustomMetricsAdapterServerOptions struct {
-	// genericoptions.ReccomendedOptions - EtcdOptions
 	SecureServing  *genericoptions.SecureServingOptionsWithLoopback
 	Authentication *genericoptions.DelegatingAuthenticationOptions
 	Authorization  *genericoptions.DelegatingAuthorizationOptions
 	Audit          *genericoptions.AuditOptions
 	Features       *genericoptions.FeatureOptions
 
-	// OpenAPIConfig
 	OpenAPIConfig *openapicommon.Config
+	EnableMetrics bool
 }
 
+// NewCustomMetricsAdapterServerOptions creates a new instance of
+// CustomMetricsAdapterServerOptions with its default values.
 func NewCustomMetricsAdapterServerOptions() *CustomMetricsAdapterServerOptions {
 	o := &CustomMetricsAdapterServerOptions{
 		SecureServing:  genericoptions.NewSecureServingOptions().WithLoopback(),
 		Authentication: genericoptions.NewDelegatingAuthenticationOptions(),
 		Authorization:  genericoptions.NewDelegatingAuthorizationOptions(),
 		Audit:          genericoptions.NewAuditOptions(),
 		Features:       genericoptions.NewFeatureOptions(),
+
+		EnableMetrics: true,
```

Comment:
Since this is a new feature that wasn't supported before, I prefer if this was added in a separate just for the sake of making it clearer in the release notes.

Reply:
Ok, I'll create a separate PR.

Reply:
Ty!